evennia/contrib/aws-s3-cdn.py

```diff
@@ -0,0 +1,1033 @@
+
+START OF SECRET_SETTINGS.PY COPY/PASTE >>>
+
+AWS_ACCESS_KEY_ID = 'THIS_IS_PROVIDED_BY_AMAZON'
+AWS_SECRET_ACCESS_KEY = 'THIS_IS_PROVIDED_BY_AMAZON'
+AWS_STORAGE_BUCKET_NAME = 'mygame-evennia' # CHANGE ME! I suggest yourgamename-evennia
+
+The settings below need to go in secret_settings,py as well, but will
+not need customization unless you want to do something particularly fancy.
+
+AWS_S3_REGION_NAME = 'us-east-1' # N. Virginia
+AWS_S3_OBJECT_PARAMETERS = { 'Expires': 'Thu, 31 Dec 2099 20:00:00 GMT',
+                            'CacheControl': 'max-age=94608000', }
+AWS_DEFAULT_ACL = 'public-read'
+AWS_S3_CUSTOM_DOMAIN = '%s.s3.amazonaws.com' % settings.AWS_BUCKET_NAME
+AWS_AUTO_CREATE_BUCKET = True
+STATICFILES_STORAGE = 'evennia.contrib.aws-s3-cdn.S3Boto3Storage'
+
+<<< END OF SECRET_SETTINGS.PY COPY/PASTE
+
+You may also store these keys as environment variables of the same name.
+For advanced configuration, refer to the docs for django-storages.
+
+UNINSTALLATION:
+
+If you haven't made changes to your static files (uploaded images, etc),
+you can simply remove the lines you added to secret_settings.py. If you
+have made changes and want to install at a later date, you can export
+your files from your S3 bucket and put them in /static/ in the evennia
+directory.
+
+LICENSE:
+
+Draws heavily from code provided by django-storages, for which these contributors
+are authors:
+
+Marty Alchin (S3)
+David Larlet (S3)
+Arne Brodowski (S3)
+Sebastian Serrano (S3)
+Andrew McClain (MogileFS)
+Rafal Jonca (FTP)
+Chris McCormick (S3 with Boto)
+Ivanov E. (Database)
+Ariel Núñez (packaging)
+Wim Leers (SymlinkOrCopy + patches)
+Michael Elsdörfer (Overwrite + PEP8 compatibility)
+Christian Klein (CouchDB)
+Rich Leland (Mosso Cloud Files)
+Jason Christa (patches)
+Adam Nelson (patches)
+Erik CW (S3 encryption)
+Axel Gembe (Hash path)
+Waldemar Kornewald (MongoDB)
+Russell Keith-Magee (Apache LibCloud patches)
+Jannis Leidel (S3 and GS with Boto)
+Andrei Coman (Azure)
+Chris Streeter (S3 with Boto)
+Josh Schneier (Fork maintainer, Bugfixes, Py3K)
+Anthony Monthe (Dropbox)
+EunPyo (Andrew) Hong (Azure)
+Michael Barrientos (S3 with Boto3)
+piglei (patches)
+Matt Braymer-Hayes (S3 with Boto3)
+Eirik Martiniussen Sylliaas (Google Cloud Storage native support)
+Jody McIntyre (Google Cloud Storage native support)
+Stanislav Kaledin (Bug fixes in SFTPStorage)
+Filip Vavera (Google Cloud MIME types support)
+Max Malysh (Dropbox large file support)
+Scott White (Google Cloud updates)
+Alex Watt (Google Cloud Storage patch)
+Jumpei Yoshimura (S3 docs)
+Jon Dufresne
+Rodrigo Gadea (Dropbox fixes)
+Martey Dodoo
+Chris Rink
+Shaung Cheng (S3 docs)
+Andrew Perry (Bug fixes in SFTPStorage)
+
+The repurposed code from django-storages is released under BSD 3-Clause,
+same as Evennia, so for detailed licensing, refer to the Evennia license.
+
+VERSIONING:
+
+This is confirmed to work for Django 2 and Django 3.
+'
```

Comment:
left-over quote character?

Reply:
necessary to fix buggy code coloring in atom IDE. if it doesn't break anybody else's IDE (assuming they properly interpret """docstrings""" , probably does more good than harm.